emulator/periph/src/flash_ctrl.rs

```diff
@@ -333,7 +350,180 @@ impl DummyFlashCtrl {
     }
 }
 
-impl FlashPeripheral for DummyFlashCtrl {
+impl MainFlashPeripheral for DummyFlashCtrl {
+    fn set_dma_ram(&mut self, ram: std::rc::Rc<std::cell::RefCell<emulator_bus::Ram>>) {
+        self.dma_ram = Some(ram);
+    }
+
+    fn poll(&mut self) {
+        if self.timer.fired(&mut self.operation_start) {
+            self.process_io();
+        }
+    }
+
+    fn read_fl_interrupt_state(
+        &mut self,
+        _size: emulator_types::RvSize,
+    ) -> emulator_bus::ReadWriteRegister<
+        u32,
+        registers_generated::main_flash_ctrl::bits::FlInterruptState::Register,
+    > {
+        emulator_bus::ReadWriteRegister::new(self.interrupt_state.reg.get())
+    }
+
+    fn write_fl_interrupt_state(
+        &mut self,
+        _size: emulator_types::RvSize,
```

Comment:
nit: It might be good to import some of these types to make the functions a little easier to read.